react/features/video-status-label/components/VideoStatusLabel.js

```diff
@@ -24,25 +24,114 @@ export class VideoStatusLabel extends Component {
         _audioOnly: React.PropTypes.bool,
 
         /**
+         * Whether or not a connection to a conference has been established.
+         */
+        _conferenceStarted: React.PropTypes.bool,
+
+        /**
          * Whether or not a high-definition large video is displayed.
          */
-        _largeVideoHD: React.PropTypes.bool
+        _largeVideoHD: React.PropTypes.bool,
+
+        /**
+         * Invoked to request toggling of audio only mode.
+         */
+        dispatch: React.PropTypes.func,
+
+        /**
+         * Invoked to obtain translated strings.
+         */
+        t: React.PropTypes.func
+    }
+
+    /**
+     * Initializes a new {@code VideoStatusLabel} instance.
+     *
+     * @param {Object} props - The read-only React Component props with which
+     * the new instance is to be initialized.
+     */
+    constructor(props) {
+        super(props);
+
+        // Bind event handler so it is only bound once for every instance.
+        this._toggleAudioOnly = this._toggleAudioOnly.bind(this);
     }
 
     /**
      * Implements React's {@link Component#render()}.
      *
      * @inheritdoc
-     * @returns {ReactElement|null}
+     * @returns {ReactElement}
      */
     render() {
-        if (this.props._audioOnly) {
-            return <AudioOnlyLabel />;
-        } else if (this.props._largeVideoHD) {
-            return <HDVideoLabel />;
+        const { _audioOnly, _conferenceStarted, _largeVideoHD, t } = this.props;
+
+        if (!_conferenceStarted || _largeVideoHD === undefined) {
```

Comment:
I almost want to say maybe you do want to rely on large video being created to show the label. The label is supposed to show what the resolution is for the stream displayed on large video. Maybe in the future large video would have the label as a child. In that sense conference started might not be needed; I think I had added it defer the label from showing while the conference was still loading.

I'm all for another state.